lib/lists.nix

```
@@ -663,9 +663,6 @@ rec {
   /* Test if two lists have no common element.
      It should be slightly more efficient than (intersectLists a b == [])
   */
-  mutuallyExclusive = a: b:
-    (builtins.length a) == 0 ||
-    (!(builtins.elem (builtins.head a) b) &&
-     mutuallyExclusive (builtins.tail a) b);
+  mutuallyExclusive = a: b: length a == 0 || length b == 0 || foldl' (p: n: p && !(elem n b)) true b;
```

this always compares `b` to itself, did you mean `elem n a`?

also it can be optimized further to stop searching on the first match and for fewer operator applications:
```suggestion
  mutuallyExclusive = a: b: length a == 0 || length b == 0 || ! builtins.any (v: builtins.elem v a) b;
```

---

You are right, I definitely overlooked this. Thanks!